packages/dds/tree/src/test/tableSchema.spec.ts

```diff
@@ -1455,16 +1458,159 @@ describe("TableFactory unit tests", () => {
 		assert.equal(column, column0);
 	});
 
+	// The following tests demonstrate more complex usage scenarios.
+	describe("User stories", () => {
+		// The following test is a demonstration of how users can add runtime type constraints for cells in a column.
+		// TODO: in the future, we want to find better ways to express such type constraints directly through the factory APIs.
+		it("Using column props for runtime type constraints", () => {
+			// The underlying storage format for the cell data is a string.
+			const Cell = schemaFactory.string;
+			type Cell = TreeNodeFromImplicitAllowedTypes<typeof Cell>;
+
+			enum ColumnType {
+				/** A raw string */
+				String = "string",
+				/** An integer - must be parsable by `Number.parseInt` */
+				Integer = "integer",
+				/** A date - must be parsable by `Date.parse` */
+				Date = "date",
+			}
+
+			const ColumnTypeNodes = adaptEnum(schemaFactory, ColumnType);
+			// Defined the types of the nodes which correspond to this the schema.
+			type ColumnTypeNodes = TreeNodeFromImplicitAllowedTypes<typeof ColumnTypeNodes.schema>;
+
+			class MyColumn extends TableSchema.column({
+				schemaFactory,
+				cell: Cell,
+				props: schemaFactory.object("TableColumnProps", {
+					/** Column label */
+					label: schemaFactory.string,
+					/** ColumnType */
+					type: ColumnTypeNodes.schema,
```

Comment:
I think it would make more sense to store the type as a schema in a static field of the column, that to store it as content in the document.

`public static readonly CellType = your schema here;`

Reply:
personally though, I don't see much point of making a nice demo for this pattern when it wouldn't be significantly harder to just build it in properly using the existing cell schema you already store on the column.

Reply:
I'll just remove this for now then. This is the pattern Nick's demo uses. I'd like to evaluate any different pattern before documenting it.

Reply:
Example removed. Will follow up separately.